sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -626,6 +626,14 @@ object SQLConf {
     .stringConf
     .createWithDefault("parquet")
 
+  val DATA_SOURCE_TABLE_INHERIT_PERMS = buildConf("spark.sql.datasource.table.inherit.perms")
+    .internal()
+    .doc("Set this to true if the table directories should be inheriting the permission " +
+      "of the warehouse or database directory " +
```

Comment:
I think this is not very clear: it seems to me that the permissions are inherited from the base directory where the table is going to be written. On the other hand, instead, the permissions are inherited from the previous table directory which was deleted, right? Can we make this more clear?

Reply:
Copy from `hive.warehouse.subdir.inherit.perms`: https://github.com/apache/hive/blob/rel/release-1.2.2/common/src/java/org/apache/hadoop/hive/conf/HiveConf.java#L1729

Reply:
yes, but that does a different thing IIUC, ie. that copies the permissions from the warehouse dir for the new tables created there. Here we are not doing the same, we are copying the permissions from the previous table dir to the new table dir after the overwrite.

Reply:
Hive configuration doc itself doesn't looks quite clear to me as well actually.